### PR TITLE
rebase to kube 1.7 on master

### PR DESCRIPTION
**Pull request by deads2k**
Rebase branch on master.  Original review of all commits completed in https://github.com/openshift/origin/pull/15100

[test]

### CODE REVIEW

**Comment by deads2k** [comment]
I think the last bit of outstanding work is to open the upstream commits.

**Comment by deads2k** [comment]
@eparis @sttts @liggitt @smarterclayton Holy crap, fully reviewed and green!  merge it quick!

**Comment by enj** [comment]
scared

**Comment by bparees** [comment]
[testextended][extended:core(builds)]

**Comment by bparees** [comment]
[testextended][extended:core(templates)]

**Comment by smarterclayton** [comment]
Can we continue to get folks fixing things merged into this until we branch?

**Comment by bparees** [comment]
@stevekuznetsov err:
```
+ make test-extended 'SUITE=core(templates),core' FOCUS=builds
test/extended/core(templates),core.sh --ginkgo.focus="builds"
/bin/sh: -c: line 0: syntax error near unexpected token `templates'
/bin/sh: -c: line 0: `test/extended/core(templates),core.sh --ginkgo.focus="builds"'
```

?

i'm going to just run one and then the other, but it seems like rosie didn't combine the focuses properly.

**Comment by deads2k** [comment]
@smarterclayton I'd rather not. This is green and trying to manually determine if their code can/should merge on mine isn't good from a review or stability perspective.  

People can create their own branches from here for merge after this.  We could also move the branch up.  I don't want to become Rosie on a psuedo-fork that periodically gets to rebase everyone's pulls on a new head.  It's hard enough keeping this up to date already

**Comment by smarterclayton** [comment]
Let me rephrase this - have you made sure everyone whose code you disabled
who *isn't* working on 3.6 closeout knows they can now start using this to
test their fixes?

**Comment by bparees** [comment]
```
The BuildConfig "sample-build" is invalid: 
* spec.source.git.httpproxy: Invalid value: "127.0.0.1:3128": proxy is not a valid url
* spec.source.git.httpsproxy: Invalid value: "127.0.0.1:3128": proxy is not a valid url
```

@jim-minter looks like we have another case of using url parse that has been broken by golang 1.8.  Think you can throw together a commit for @deads2k to cherry-pick here?

File: pkg/router/metrics/haproxy/haproxy.go [hunk 1]
```diff
@@ -415,8 +415,9 @@ loop:
 	}
 
 	e.serverLimited = servers > e.opts.ServerThreshold
-	e.serverThresholdCurrent.Set(float64(servers))
-	e.serverThresholdLimit.Set(float64(e.opts.ServerThreshold))
+	// TODO counters only go up
+	// e.serverThresholdCurrent.Set(float64(servers))
```
Comment (smarterclayton) [inline pkg/router/metrics/haproxy/haproxy.go]: You can just make these gauges

**Comment by smarterclayton** [comment]
Saw this in the logs:

```
W0718 03:42:32.685569   20120 reflector.go:323] github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/client/informers/informers_generated/externalversions/factory.go:72: watch of *v1beta1.ThirdPartyResource ended with: etcdserver: mvcc: required revision has been compacted
```

GC or lost Mo's "disable TPR controller" commit?

**Comment by liggitt** [comment]
> W0718 03:42:32.685569   20120 reflector.go:323] github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/client/informers/informers_generated/externalversions/factory.go:72: watch of *v1beta1.ThirdPartyResource ended with: etcdserver: mvcc: required revision has been compacted

could also be storage cacher... not sure

**Comment by smarterclayton** [comment]
How are you tracking post-rebase (so I can add things)

**Comment by sttts** [comment]
@smarterclayton post rebase todos: https://github.com/openshift/origin/issues/14647

**Comment by stevekuznetsov** [comment]
@bparees that logic lives [here](https://github.com/openshift/aos-cd-jobs/blob/master/sjb/config/test_cases/test_branch_origin_extended.yml#L18-L24)

**Comment by smarterclayton** [comment]
etcd regressed to v3.1.5, needs to be 3.2.1

**Comment by smarterclayton** [comment]
Opened a PR for etcd

**Comment by deads2k** [comment]
> Opened a PR for etcd

It didn't build.  I've opened https://github.com/openshift/origin/issues/15313 and I'll keep chasing it.

**Comment by deads2k** [comment]
[test]

**Comment by deads2k** [comment]
All the upstream 00000's have associated pulls.

**Comment by deads2k** [comment]
I kicked a few jobs for flake handling: 

1. https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin/3233/
1. https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin/3234/
1. https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin/3235/
1. https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin/3236/

**Comment by sttts** [comment]
🎉
